core/lib/config/Cargo.toml

```diff
@@ -12,7 +12,13 @@ categories = ["cryptography"]
 [dependencies]
 zksync_basic_types = { path = "../../lib/basic_types" }
 zksync_contracts = { path = "../../lib/contracts" }
+zksync_protobuf.workspace = true
```

Comment:
I'm afraid this kills the idea of detaching config definitions from the serialization format. Chances are, we'll move from depending on env variables to be always present to depending on protobuf configs to be always present, which is not really better.
I'd assume that we need a proxy crate like `zksync_protobuf_config` that will only be used in binary crates, so that library code has no access to some kind of "default" config provider.

Reply:
I've created a separate crate and defined a trait ProtoRepr (reverse to ProtoFmt) since rust does not allow to implement an external trait for external type. The trait and auxiliary encode/decode methods are currently defined locally but will be moved to zksync_protobuf crate later.